### PR TITLE
[15.0][MIG] stock_picking_product_barcode_report: Migration to 15.0

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT40405

please @sergio-teruel @carlosdauden review this

### CODE REVIEW

**Review by sergio-teruel — APPROVED** [review]
Functional and code review

**Comment by sergio-teruel** [comment]
pre-commit fixed here https://github.com/OCA/stock-logistics-barcode/pull/464

File: stock_picking_product_barcode_report/models/res_company.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2021 Tecnativa - Carlos Roca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class Company(models.Model):
+    _inherit = "res.company"
+
+    barcode_default_format = fields.Selection(
```
Comment (rousseldenis) [inline stock_picking_product_barcode_report/models/res_company.py]: @CarlosRoca13 @pedrobaeza Shouldn't this defined in a more generic module ?

Comment (pedrobaeza) [reply stock_picking_product_barcode_report/models/res_company.py]: Well, it's specific for this report, so it doesn't make sense in a generic one.

**Comment by chienandalu** [comment]
Please rebase @CarlosRoca13

**Comment by CarlosRoca13** [comment]
Done @chienandalu

File: stock_picking_product_barcode_report/models/res_company.py (hunk 1 above)
Comment (rousseldenis) [reply stock_picking_product_barcode_report/models/res_company.py]: So, maybe the field name too generic is.

Comment (pedrobaeza) [reply stock_picking_product_barcode_report/models/res_company.py]: Very Yoda your sentence is, hehe. @CarlosRoca13 please change it to `barcode_report_default_format`.

**Comment by CarlosRoca13** [comment]
Done! I add the migration script to rename the field too. 

BTW I don't know if the field renaming of `res.config.settings` is needed

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**